File: src/test/java/com/raychenon/leetcode/MergeIntervalsTest.java
package com.raychenon.leetcode;

import org.junit.BeforeClass;
import org.junit.Test;

import java.util.Arrays;

import static junit.framework.TestCase.assertTrue;

/**
 * User: raychenon
 * Date: 2020-02-02
 * https://leetcode.com/problems/merge-intervals/
 */
public class MergeIntervalsTest {

    private static MergeIntervals instance;

    @BeforeClass
    public static void setUp() {
        instance = new MergeIntervals();
    }


    @Test
    public void example1Test() {
        int input[][] = {{1, 3}, {2, 6}, {8, 10}, {15, 18}};
        int expectedResult[][] = {{1, 6}, {8, 10}, {15, 18}};
        assertTrue(Arrays.deepEquals(expectedResult, instance.merge(input)));
    }

    @Test
    public void example2Test() {
        int input[][] = {{1, 4}, {4, 5}};
        int expectedResult[][] = {{1, 5}};
        assertTrue(Arrays.deepEquals(expectedResult, instance.merge(input)));
    }

    @Test
    public void example3Test() {
        int input[][] = {{1, 4}, {0, 4}};
        int expectedResult[][] = {{0, 4}};
        assertTrue(Arrays.deepEquals(expectedResult, instance.merge(input)));
    }

    @Test
    public void example4Test() {
        int input[][] = {{1, 4}, {0, 1}};
        int expectedResult[][] = {{0, 4}};
        assertTrue(Arrays.deepEquals(expectedResult, instance.merge(input)));
    }

    @Test
    public void example5Test() {
        int input[][] = {{1, 4}, {0, 0}};
        int expectedResult[][] = {{0, 0}, {1, 4}};
        assertTrue(Arrays.deepEquals(expectedResult, instance.merge(input)));
    }

    @Test
    public void emptyIntervalTest() {
        int input[][] = {};
        int expectedResult[][] = {};
        assertTrue(Arrays.deepEquals(expectedResult, instance.merge(input)));
    }
}

File: src/test/kotlin/com/raychenon/kotlin/leetcode/MergeIntervalsTest.kt
package com.raychenon.kotlin.leetcode

import junit.framework.TestCase
import org.junit.Test
import java.util.Arrays

/**
 * User: raychenon
 * Date: 2020-01-30
 * https://leetcode.com/problems/merge-intervals/
 */
class MergeIntervalsTest {

    @Test
    fun example1Test() {
        val input = arrayOf(intArrayOf(1, 3), intArrayOf(2, 6), intArrayOf(8, 10), intArrayOf(15, 18))
        TestCase.assertTrue(
            Arrays.deepEquals(
                arrayOf(intArrayOf(1, 6), intArrayOf(8, 10), intArrayOf(15, 18)),
                MergeIntervals.merge(input)
            )
        )
    }

    @Test
    fun example2Test() {
        val input = arrayOf(intArrayOf(1, 4), intArrayOf(4, 5))
        TestCase.assertTrue(Arrays.deepEquals(arrayOf(intArrayOf(1, 5)), MergeIntervals.merge(input)))
    }

    @Test
    fun example3Test() {
        val input = arrayOf(intArrayOf(1, 4), intArrayOf(0, 4))
        TestCase.assertTrue(Arrays.deepEquals(arrayOf(intArrayOf(0, 4)), MergeIntervals.merge(input)))
    }

    @Test
    fun example4Test() {
        val input = arrayOf(intArrayOf(1, 4), intArrayOf(0, 1))
        TestCase.assertTrue(Arrays.deepEquals(arrayOf(intArrayOf(0, 4)), MergeIntervals.merge(input)))
    }

    @Test
    fun example5Test() {
        val input = arrayOf(intArrayOf(1, 4), intArrayOf(0, 0))
        val expectedResult = arrayOf(intArrayOf(0, 0), intArrayOf(1, 4))
        TestCase.assertTrue(Arrays.deepEquals(expectedResult, MergeIntervals.merge(input)))
    }

    @Test
    fun emptyArray_test() {
        TestCase.assertTrue(Arrays.deepEquals(emptyArray<IntArray>(), MergeIntervals.merge(emptyArray<IntArray>())))
    }
}